wondrous-app/components/organization/collaborations/CollaborationItem.tsx

```diff
@@ -0,0 +1,199 @@
+import { SharedOrgHeaderCard } from 'components/Collaboration/SharedOrgHeader';
+import RolePill from 'components/Common/RolePill';
+import { Modal as ModalComponent } from 'components/Modal';
+import SmartLink from 'components/Common/SmartLink';
+import TaskCardPrivacy from 'components/Common/TaskCardPrivacy';
+import { useContext, useEffect, useState } from 'react';
+import { PERMISSIONS } from 'utils/constants';
+import { parseUserPermissionContext } from 'utils/helpers';
+import { useMutation } from '@apollo/client';
+import { APPROVE_ORG_COLLAB_REQUEST, DECLINE_ORG_COLLAB_REQUEST } from 'graphql/mutations';
+import CollabDetails, { MODAL_TYPE } from 'components/CreateCollaborationModal/ViewCollab/CollabDetails';
+import { SnackbarAlertContext } from 'components/Common/SnackbarAlert';
+import { Actions } from 'components/CreateCollaborationModal/ViewCollab';
+import {
+  CollabBottom,
+  CollabCard,
+  CollabCardHeader,
+  CollabDescription,
+  InvitationButton,
+  InvitationButtonText,
+  StyledBottomHr,
+} from './styles';
+import { RoleButtonWrapper } from '../wrapper/styles';
+
+const ORG_PERMISSIONS = {
+  MANAGE_SETTINGS: 'manageSettings',
+  CONTRIBUTOR: 'contributor',
+};
+
+export const ActiveCollaborationItem = (props) => {
+  const { collab, userPermissionsContext } = props;
+  const [orgRoleName, setOrgRoleName] = useState(null);
+  const [permissions, setPermissions] = useState(undefined);
+  useEffect(() => {
```

Comment:
I wonder if we can avoid state & effect here and just have memoized values. So something like 

` const orgPermissions = parseUserPermissionContext({
      userPermissionsContext,
      orgId: collab?.id,
    });


  const permissions = useMemo(() => {
    if (
      orgPermissions?.includes(PERMISSIONS.MANAGE_MEMBER) ||
      orgPermissions?.includes(PERMISSIONS.FULL_ACCESS) ||
      orgPermissions?.includes(PERMISSIONS.APPROVE_PAYMENT)
    ) {
      return ORG_PERMISSIONS.MANAGE_SETTINGS;
    } else if (
      userPermissionsContext?.orgPermissions &&
      collab?.id in userPermissionsContext?.orgPermissions &&
      orgPermissions
    ) {
      // Normal contributor with no access to admin settings
      return ORG_PERMISSIONS.CONTRIBUTOR
    }
  }, [orgPermissions, userPermissionsContext, collab?.id]);
  const orgRoleName = useMemo(() => userPermissionsContext?.orgRoles?[collab?.id], [collab?.id, userPermissionsContext]);`